gobblin-service/src/test/java/org/apache/gobblin/service/monitoring/FsFlowGraphMonitorTest.java

```diff
@@ -0,0 +1,310 @@
+  private static final String NODE_2_FILE = "node2.properties";
+  private final File node2Dir = new File(FileUtils.getTempDirectory(), "node2");
+  private final File node2File = new File(node2Dir, NODE_2_FILE);
+  private final File edge1Dir = new File(node1Dir, "node2");
+  private final File edge1File = new File(edge1Dir, "edge1.properties");
+
+  private RefSpec masterRefSpec = new RefSpec("master");
+  private Optional<FSFlowTemplateCatalog> flowCatalog;
+  private Config config;
+  private BaseFlowGraph flowGraph;
+  private FsFlowGraphMonitor flowGraphMonitor;
+  private Map<URI, TopologySpec> topologySpecMap;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    cleanUpDir(TEST_DIR.toString());
+
+
+    URI topologyCatalogUri = this.getClass().getClassLoader().getResource("topologyspec_catalog").toURI();
+    this.topologySpecMap = MultiHopFlowCompilerTest.buildTopologySpecMap(topologyCatalogUri);
+
+    this.config = ConfigBuilder.create()
+        .addPrimitive(FsFlowGraphMonitor.FS_FLOWGRAPH_MONITOR_PREFIX + "."
+            + ConfigurationKeys.FLOWGRAPH_REPO_DIR, this.flowGraphDir.getAbsolutePath())
+        .addPrimitive(FsFlowGraphMonitor.FS_FLOWGRAPH_MONITOR_PREFIX + "." + ConfigurationKeys.FLOWGRAPH_BASE_DIR, "gobblin-flowgraph")
+        .addPrimitive(FsFlowGraphMonitor.FS_FLOWGRAPH_MONITOR_PREFIX + "." + ConfigurationKeys.FLOWGRAPH_POLLING_INTERVAL, 2)
+        .build();
+
+    // Create a FSFlowTemplateCatalog instance
+    URI flowTemplateCatalogUri = this.getClass().getClassLoader().getResource("template_catalog").toURI();
+    Properties properties = new Properties();
+    this.flowGraphDir.mkdirs();
+    properties.put(ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY, flowTemplateCatalogUri.toString());
+    Config config = ConfigFactory.parseProperties(properties);
+    Config templateCatalogCfg = config
+        .withValue(ConfigurationKeys.JOB_CONFIG_FILE_GENERAL_PATH_KEY,
+            config.getValue(ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY));
+    this.flowCatalog = Optional.of(new FSFlowTemplateCatalog(templateCatalogCfg));
+
+    //Create a FlowGraph instance with defaults
+    this.flowGraph = new BaseFlowGraph();
+
+    this.flowGraphMonitor = new FsFlowGraphMonitor(this.config, this.flowCatalog, this.flowGraph, topologySpecMap, new CountDownLatch(1));
+    this.flowGraphMonitor.startUp();
+    this.flowGraphMonitor.setActive(true);
+  }
+
+  @Test
+  public void testAddNode() throws Exception {
+    String file1Contents = FlowGraphConfigurationKeys.DATA_NODE_IS_ACTIVE_KEY + "=true\nparam1=value1\n";
+    String file2Contents = FlowGraphConfigurationKeys.DATA_NODE_IS_ACTIVE_KEY + "=true\nparam2=value2\n";
+
+    addNode(this.node1Dir, this.node1File, file1Contents);
+    addNode(this.node2Dir, this.node2File, file2Contents);
+
+    // Let the monitor pick up the nodes that were recently added
+    Thread.sleep(3000);
+    for (int i = 0; i < 1; i++) {
```

Comment:
apologies if I'm being dense, but how do you verify `"param2=value2\n"`, amidst `i < 1` as the looping condition?

Reply:
Ah shoot I think I set the loop differently to test something else but forgot to fix, thanks for the callout.